modules/core/src/directx.cpp

```diff
@@ -1392,11 +1393,201 @@ void convertFromD3D11Texture2D(ID3D11Texture2D* pD3D11Texture2D, OutputArray dst
+    }
+    if (platform == nullptr) {
+        CV_Error(cv::Error::StsNullPtr, "OpenCL: clGetDeviceInfo returns empty platform");
+    }
+    auto&& platformName = PlatformInfo(&platform).name();
+
+    ID3D11Device *pD3D11Device = nullptr;
+    pD3D11Texture2D->GetDevice(&pD3D11Device);
+
+    if (pD3D11Device == nullptr) {
+        CV_Error(cv::Error::StsNullPtr, "OpenCL: GetDevice returns empty pD3D11Device in convertFromD3D11Texture2DtoCLMem");
+    }
+
+    cl_context context = NULL;
+    cl_context_properties properties[] = {
+            CL_CONTEXT_PLATFORM, (cl_context_properties)platform,
+            CL_CONTEXT_D3D11_DEVICE_KHR, (cl_context_properties)(pD3D11Device),
+            CL_CONTEXT_INTEROP_USER_SYNC, CL_FALSE,
+            NULL, NULL
+    };
+    context = clCreateContext(properties, 1, &cl_device, NULL, NULL, &status);
+    if (status != CL_SUCCESS && cl_device != nullptr)
+    {
+        CV_LOG_DEBUG(NULL, "clCreateContext failed with status " << int(status));
+        clReleaseDevice(cl_device);
+    }
+
+    OpenCLExecutionContext clExecCtx;
+    try
+    {
+        if (cl_device != nullptr) {
+            out_size = 0;
+            clGetDeviceInfo(cl_device, CL_DEVICE_EXTENSIONS, 0, nullptr, &out_size);
+            if (out_size <= 0) {
+                CV_Error(cv::Error::OpenCLApiCallError, "OpenCL: clGetDeviceInfo doesn't find any extensions");
+            }
+            std::string ext;
+            ext.resize(out_size, '\0');
+            clGetDeviceInfo(cl_device, CL_DEVICE_EXTENSIONS, out_size, &ext.front(), nullptr);
+            if (ext.find("cl_khr_d3d11_sharing") == std::string::npos) {
+                CV_Error(cv::Error::OpenCLApiCallError, "OpenCL: Device doesn't contain cl_khr_d3d11_sharing extension");
+            }
+        }
+        clExecCtx = OpenCLExecutionContext::create(platformName, platform, context, cl_device);
+        clExecCtx.getContext().setUserContext(std::make_shared<OpenCL_D3D11>(platform, pD3D11Device));
+    }
+    catch (...)
+    {
+        if (cl_device != nullptr) clReleaseDevice(cl_device);
+        if (context != nullptr) clReleaseContext(context);
+        throw;
+    }
+    clExecCtx.bind();
+    auto ctx = const_cast<Context&>(clExecCtx.getContext());
+
+    OpenCL_D3D11* impl = ctx.getUserContext<OpenCL_D3D11>().get();
+    if (nullptr == impl) {
+        CV_Error(cv::Error::OpenCLApiCallError, "OpenCL: Context initialized without DirectX interoperability");
+    }
+
+    cl_mem image2d = impl->clCreateFromD3D11Texture2DKHR((cl_context)ctx.ptr(),
+                                                         CL_MEM_READ_WRITE,
+                                                         pD3D11Texture2D,
+                                                         0,
+                                                         &status);
+    auto image = Image2D::fromHandle(image2d);
+    clRetainMemObject(image2d);
+    if (status != CL_SUCCESS) {
```

Comment:
please move  this check after `clCreateFromD3D11Texture2DKHR`

Reply:
Fixed.